Adding a way to find Ogre library

shrit opened the pull request:
This patch resolve the problem of finding ogre cmake files.
This problem is reported when using ogre 1.9 on arch linux

shrit committed:
Adding a way to find Ogre library

This patch resolve the problem of finding ogre cmake files.
This problem is reported when using ogre 1.9 on arch linux

TSC21 reviewed (changes requested):
@shrit I am ok with this as long as we filter it by platform, meaning there's a check for the OS and based on that, we append those paths. You can, for example, use `EXECUTE_PROCESS()` with `lsb_release -is` to check the underlying system if it is a Linux one.

julianoes commented on CMakeLists.txt:
```diff
@@ -17,6 +17,8 @@ include(GNUInstallDirs)
 #######################
 
 # Add search directory for CMake on OS X
+list(APPEND CMAKE_PREFIX_PATH "/usr/lib/OGRE/OGRE/cmake")
```
I think you need to fix or move the comment above this, unless your fix is for macOS?

shrit replied:
Exactly, it is for Arch Linux. I did not pay attention to the comment. However, at the end I need to use `uname -a` with `EXECUTE_PROCESS()` to parse the kernel name and be sure that it is  Arch Linux. Because usually` lsb_release -is ` is only for Ubuntu

julianoes replied:
I remember adding something like `PKG_CONFIG=/opt/OGRE-1.9` as an env variable in Arch Linux and that did the job.

shrit replied:
Closed in favor of https://github.com/PX4/sitl_gazebo/pull/304

shrit committed:
Merge branch 'master' of https://github.com/PX4/sitl_gazebo into ogre

shrit committed:
Add a Ogre depedency under Arch linux

Since lsb_release does not exist in arch linux, the only
way to verify the OS is by parsing uname.

Signed-off-by: Omar Shrit <shrit@lri.fr>
